lib/galaxy/managers/hdcas.py

```diff
@@ -42,6 +42,9 @@ def stream_dataset_collection(dataset_collection_instance, upstream_mod_zip=Fals
 def write_dataset_collection(dataset_collection_instance, archive):
     names, hdas = get_hda_and_element_identifiers(dataset_collection_instance)
     for name, hda in zip(names, hdas):
+        if not hda:
```

Comment:
Can you instead check that the collection instance is populated ?

Reply:
I suspected there was a much better way, but didn't know where to look. Thanks!